superdesk/io/feeding_services/ftp.py

```diff
@@ -69,74 +71,120 @@ def _test(self, provider):
         except Exception as ex:
             raise IngestFtpError.ftpError(ex, provider)
 
+    def _move(self, ftp, src, dest):
+        """Move distant file
+
+        :param ftp: FTP instance to use
+        :type ftp: ftplib.FTP
+        :param src: source path of the file to move
+        :type src: str
+        :param dest: dest path of the file to move
+        :type dest: str
+        """
+        try:
+            ftp.rename(src, dest)
+        except ftplib.all_errors as e:
+            logger.warning("Can't move file from {src} to {dest}: {reason}".format(
+                src=src,
+                dest=dest,
+                reason=e))
+
+    def _create_if_missing(self, ftp, path):
+        """Check if a dir exists, and create it else
+
+        :param ftp: FTP instance to use
+        :type ftp: ftplib.FTP
+        :param src: dir path to check
+        """
+        try:
+            next(ftp.mlsd(path))
```

Comment:
@petrjasek creating the dir directly is not possible because we don't have conflict error, but more generic ones, so we can't distinguish a conflict (which would be ignored) for existing path from a setting/permission error (which should make the moving fail).

An error code is returned which can probably be used, but it would need extensive test to be sure that most FTP server return the same one, it's a lot of complication when the current workflow works OK.